flutter/lib/desktop/pages/connection_page.dart

```diff
@@ -171,36 +234,142 @@ class _ConnectionPageState extends State<ConnectionPage>
-                      focusNode: _idFocusNode,
-                      style: const TextStyle(
-                        fontFamily: 'WorkSans',
-                        fontSize: 22,
-                        height: 1.4,
-                      ),
-                      maxLines: 1,
-                      cursorColor:
-                          Theme.of(context).textTheme.titleLarge?.color,
-                      decoration: InputDecoration(
-                          filled: false,
-                          counterText: '',
-                          hintText: _idInputFocused.value
-                              ? null
-                              : translate('Enter Remote ID'),
-                          contentPadding: const EdgeInsets.symmetric(
-                              horizontal: 15, vertical: 13)),
-                      controller: _idController,
-                      inputFormatters: [IDTextInputFormatter()],
-                      onSubmitted: (s) {
-                        onConnect();
-                      },
-                    ),
-                  ),
+                  child: 
+                   Autocomplete<Peer>(
+                    optionsBuilder: (TextEditingValue textEditingValue) {
+                      if (textEditingValue.text == '') {
+                        return const Iterable<Peer>.empty();
+                      }
+                      else if (peers.isEmpty) {
+                         Peer emptyPeer = Peer(
+                          id: '',
+                          username: '',
+                          hostname: '',
+                          alias: '',
+                          platform: '',
+                          tags: [],
+                          hash: '',
+                          forceAlwaysRelay: false,
+                          rdpPort: '',
+                          rdpUsername: '',
+                          loginName: '',
+                        );
+                        return [emptyPeer];
+                      }
+                      else {
+                        String textWithoutSpaces = textEditingValue.text.replaceAll(" ", "");
+                        if (int.tryParse(textWithoutSpaces) != null) {
+                          textEditingValue = TextEditingValue(
+                            text: textWithoutSpaces,
+                            selection: textEditingValue.selection,
+                          );
+                        }
+
+                        return peers.where((peer) =>
+                        peer.id.toLowerCase().contains(textEditingValue.text.toLowerCase()) ||
+                        peer.username.toLowerCase().contains(textEditingValue.text.toLowerCase()) ||
+                        peer.hostname.toLowerCase().contains(textEditingValue.text.toLowerCase()) ||
+                        peer.alias.toLowerCase().contains(textEditingValue.text.toLowerCase()))
+                            .toList();
```

Comment:
Improve here. no need to call textEditingValue.text.toLowerCase() in every where.